lib/scala/kg/src/main/scala/io/github/tetherlessworld/mcsapps/lib/kg/stores/postgres/PostgresKgQueryStore.scala

```diff
@@ -1,29 +1,64 @@
 package io.github.tetherlessworld.mcsapps.lib.kg.stores.postgres
 
 import com.google.inject.Inject
+import io.github.tetherlessworld.mcsapps.lib.kg.models.edge.KgEdge
 import io.github.tetherlessworld.mcsapps.lib.kg.models.node.{KgNode, KgNodeContext, KgNodeLabel, KgNodeLabelContext}
 import io.github.tetherlessworld.mcsapps.lib.kg.models.path.KgPath
 import io.github.tetherlessworld.mcsapps.lib.kg.models.search.{KgSearchFacets, KgSearchQuery, KgSearchResult, KgSearchSort}
 import io.github.tetherlessworld.mcsapps.lib.kg.models.source.KgSource
 import io.github.tetherlessworld.mcsapps.lib.kg.stores.KgQueryStore
 import javax.inject.Singleton
+import slick.jdbc.GetResult
 
 import scala.concurrent.ExecutionContext
 
 @Singleton
 final class PostgresKgQueryStore @Inject()(configProvider: PostgresStoreConfigProvider)(implicit executionContext: ExecutionContext) extends AbstractPostgresKgStore(configProvider) with KgQueryStore {
   import profile.api._
 
+  private implicit val getStringList = GetResult[List[String]] (r =>
+    r.rs.getArray(r.skip.currentPos)
+      .getArray
+      .asInstanceOf[Array[Any]]
+      .toList
+      .map(_.toString())
+  )
+
+  private implicit val getKgEdge = GetResult(r => KgEdge(
```

Comment:
I'm not familiar with this syntax, but I'd guess it's tying you to a column order? which is not ideal. Should usually get columns by name and not by position.

Reply:
Yeah it was tied to column order, changed to use column names instead